### DISCUSSION TITLE
[@types/chart.js] issues with ChartScales

### DISCUSSION THREAD

**Discussion by burlap101** [DefinitelyTyped/DefinitelyTyped#55264] [Issues with a @types package] [answered]
Hi i am having issues reconciling the type definitions for the scales property of a `Chart.ChartOptions` object and the Chart.js documentation. I feel there is a problem with the type definition rather than with the docs. 

An example object from the [docs](https://www.chartjs.org/docs/master/axes/cartesian/#creating-multiple-axes) :
```javascript
options: {
        scales: {
            'left-y-axis': {
                type: 'linear',
                position: 'left'
            },
            'right-y-axis': {
                type: 'linear',
                position: 'right'
            }
        }
    }
```

which doesn't match the interface for the [ChartOptions.scale](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e6491e0d87a72a566c0f6ce61fca2b57199aa172/types/chart.js/index.d.ts#L315) property which is defined as a [Chart.ChartScales](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/e6491e0d87a72a566c0f6ce61fca2b57199aa172/types/chart.js/index.d.ts#L751) type:
```typescript
interface ChartScales {
        type?: ScaleType | string | undefined;
        display?: boolean | undefined;
        position?: PositionType | string | undefined;
        gridLines?: GridLineOptions | undefined;
        scaleLabel?: ScaleTitleOptions | undefined;
        ticks?: TickOptions | undefined;
        xAxes?: ChartXAxe[] | undefined;
        yAxes?: ChartYAxe[] | undefined;
    }
```

[a comment by an automated account is not shown]

**Comment by lekoaf** [comment, marked as answer]
You're looking a the master documentation. Since 3.X of Chart.js they write their own type definitions. So this library only supports types up to 2.9.X. Double check the docs to compare with 2.9.X instead.

**Reply by burlap101** [reply]
yep thanks @lekoaf that was where i was hitting the wall. have now been looking at type definitions in the chart.js repo [here](https://github.com/chartjs/Chart.js/blob/master/types/index.esm.d.ts).

**Reply by mackuba** [reply]
@lekoaf I'm having trouble using those new type definitions, because they're written in a different way (as modules, not in global scope)… Do you have any idea if it's possible to use them same way as the old ones without any changes on their side, e.g. with some kind of wrapper or something? I've described the problem here: https://github.com/chartjs/Chart.js/issues/9965

I'm honestly kind of lost because I haven't used JS modules before, and I don't want to get that invested in the JS world to learn this right now, but I'd really love to upgrade to 3.0 because with 2.9 the charts work really slow in Chrome. So I'm facing a choice of either keeping them slow or converting the code back to plain JS and dropping the types, on which I've spent some time setting up…

**Reply by lekoaf** [reply]
Sorry, haven't actually tried 3.x of Chart.js myself yet. The change of api and types was too much of a hazzle to upgrade.
